src/EventLoop.cpp

```diff
@@ -282,12 +281,14 @@ auto EventLoop::checkActions(
             CURLcode easy_result = message->data.result;
 
             // So the reacquired shared_ptr destructs itself correctly, let's get it into a unique pointer
-            std::unique_ptr<std::shared_ptr<SharedRequest>> shared_request_ptr_pointer;
-            curl_easy_getinfo(easy_handle, CURLINFO_PRIVATE, &shared_request_ptr_pointer);
+            std::shared_ptr<SharedRequest>* curl_info_private_pointer{nullptr};
+            curl_easy_getinfo(easy_handle, CURLINFO_PRIVATE, &curl_info_private_pointer);
             curl_multi_remove_handle(m_cmh, easy_handle);
 
+            std::unique_ptr<std::shared_ptr<SharedRequest>> shared_request_ptr_pointer{curl_info_private_pointer};
+
             auto shared_request_ptr = *shared_request_ptr_pointer;
-            
+
```

Comment:
Should reduce the double call below:
`shared_request_ptr->GetAsReference()`

Reply:
Good idea